model/ftn/w3nmluprstrmd.ftn

```diff
@@ -0,0 +1,529 @@
+    CHARACTER(15)               :: RESTARTTIME
+  END TYPE NML_RESTART_T
+
+  ! update approach
+  TYPE NML_UPDATE_T
+    CHARACTER(5)                :: UPDPROC
+    REAL                        :: PRCNTG
+    REAL                        :: PRCNTGCAP
+    REAL                        :: THRWSEA
+    CHARACTER(30)               :: FILE
+  END TYPE NML_UPDATE_T
+
+  ! miscellaneous
+  CHARACTER(256)                :: MSG
+  INTEGER                       :: NDSN
+
+
+
+
+  CONTAINS
+!/ ------------------------------------------------------------------- /
+  SUBROUTINE W3NMLUPRSTR (NDSI, INFILE, NML_RESTART, NML_UPDATE, IERR)
+!/
+!/                  +-----------------------------------+
+!/                  | WAVEWATCH III           NOAA/NCEP |
+!/                  |           M. Accensi              |
+!/                  |                                   |
+!/                  |                        FORTRAN 90 |
+!/                  | Last update :         06-Oct-2020 |
+!/                  +-----------------------------------+
+!/
+!
+!  1. Purpose :
+!
+!     Reads all the namelist to define the output field
+!
+!  2. Method :
+!
+!     See source term routines.
+!
+!  3. Parameters :
+!
+!     Parameter list
+!     ----------------------------------------------------------------
+!      NDSI        Int.
+!      INFILE      Char.
+!      NML_RESTART type.
+!      NML_UPDATE  type.
+!      IERR        Int.
+!     ----------------------------------------------------------------
+!
+!  4. Subroutines used :
+!
+!      Name      TYPE  Module   Description
+!     ----------------------------------------------------------------
+!      STRACE    Subr. W3SERVMD SUBROUTINE tracing.
+!      READ_RESTART_NML
+!      READ_UPDATE_NML
+!     ----------------------------------------------------------------
+!
+!  5. Called by :
+!
+!      Name       TYPE   Module   Description
+!     ----------------------------------------------------------------
+!      WW3_UPRSTR Prog.  N/A      Update restart file
+!     ----------------------------------------------------------------
+!
+!  6. Error messages :
+!
+!     None.
+!
+!  7. Remarks :
+!
+!  8. Structure :
+!
+!     See source code.
+!
+!  9. Switches :
+!
+! 10. Source code :
+!
+!/ ------------------------------------------------------------------- /
+
+    USE W3ODATMD, ONLY: NDSE
+!/S      USE W3SERVMD, ONLY: STRACE
+
+    IMPLICIT NONE
+
+    INTEGER, INTENT(IN)                         :: NDSI
+    CHARACTER*(*), INTENT(IN)                   :: INFILE
+    TYPE(NML_RESTART_T), INTENT(INOUT)          :: NML_RESTART
+    TYPE(NML_UPDATE_T), INTENT(INOUT)           :: NML_UPDATE
+    INTEGER, INTENT(OUT)                        :: IERR
+!/S      INTEGER, SAVE                             :: IENT = 0
+
+    IERR = 0
+!/S      CALL STRACE (IENT, 'W3NMLUPRSTR')
+
+    ! open namelist log file
+    NDSN = 3
+    OPEN (NDSN, file=TRIM(INFILE)//'.log', form='formatted', iostat=IERR)
+       IF (IERR.NE.0) THEN
```

Comment:
Does this indent before 'If' need to be removed? (Unlikely to make any difference to code, but just an aesthetic change.)

Reply:
Agreed and done!